### PR TITLE
Groups: Introduction banner and empty state in persons page

### PR DESCRIPTION
**Pull request by macobo**
## Changes

This PR adds groups introduction/upsell prompts with accordance of https://github.com/PostHog/posthog/issues/7491

The prompts are behind a separate feature flag given docs are not yet ready and to avoid internal confusion.

Note for reviewer: I'll be building on top of this PR so will resolve comments in a follow-up. :)

## Screenshots

image
Banner, user has access to group analytics

image
Banner, when user has instrumented their code but not upgraded

image
Banner, user needs to upgrade + instrument

image
Persons empty state, user needs to upgrade + instrument

image
Persons empty state, when user has instrumented their code but not upgraded

Etc..

## How did you test this code?

See various screenshots

### CODE REVIEW

File: frontend/src/scenes/groups/GroupsTabs.tsx [hunk 1]
```diff
@@ -9,13 +10,25 @@ export function GroupsTabs(): JSX.Element {
     const { setTab } = useActions(groupsListLogic)
     const { currentTab } = useValues(groupsListLogic)
     const { groupTypes } = useValues(groupsModel)
+    const { groupsAccessStatus } = useValues(groupsAccessLogic)
+
+    const showGroupsIntroductionPage = [
```
Comment (paolodamico) [inline frontend/src/scenes/groups/GroupsTabs.tsx:15]: This is confusing, it makes it look like we show this when for instance the user has no access to groups at all but the `GroupsTabs` component is only mounted when groups are fully enabled and further, at least one group exists, so is this actually being used at all?

File: frontend/src/scenes/groups/GroupsIntroduction.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    access: GroupsAccessStatus.NoAccess | GroupsAccessStatus.HasAccess | GroupsAccessStatus.HasGroupTypes
+}
+
+export function GroupsIntroduction({ access }: Props): JSX.Element {
+    let title, subtext, primaryButton, secondaryButton
+
+    const { upgradeLink } = useValues(groupsAccessLogic)
+
+    const upgradeButton = (
+        <LinkButton
```
Comment (paolodamico) [inline frontend/src/scenes/groups/GroupsIntroduction.tsx:18]: Nit: Let's use the new Lemon Buttons here.

File: frontend/src/scenes/groups/GroupsIntroduction.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    } else {
+        // HasGroupTypes
+        title = <>Looks like you're tracking groups!</>
+        subtext = <>Upgrade today to use groups in Insights.</>
+        primaryButton = upgradeButton
+    }
+
+    return (
+        <div className="groups-introduction">
+            <h2>{title}</h2>
```
Comment (paolodamico) [inline frontend/src/scenes/groups/GroupsIntroduction.tsx:67]: This should be an h3 and smaller than the current page title to follow good page hierarchy

File: frontend/src/layout/navigation/TopBar/TopBar.scss [hunk 4]
```diff
@@ -75,6 +74,17 @@
     a:visited {
         color: var(--purple-light);
     }
+    .bold {
```
Comment (paolodamico) [inline frontend/src/layout/navigation/TopBar/TopBar.scss:77]: Nit: Didn't see a change that uses this class, if possible using `strong` or `b` seems like a better approach too

File: frontend/src/lib/introductions/GroupsIntroductionBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    let introductionSegment = (
+        <>
+            <strong>🎉 Introducing group analytics!</strong> Analyze how groups interact with your product as a whole.
+        </>
+    )
+    if (groupsAccessStatus === GroupsAccessStatus.HasGroupTypes) {
+        introductionSegment = (
+            <>
+                <strong>🎉 Looks like you're tracking groups!</strong> Upgrade today to use groups in Insights.
```
Comment (paolodamico) [inline frontend/src/lib/introductions/GroupsIntroductionBanner.tsx:22]:
```suggestion
                <strong>🎉 Looks like you're tracking groups!</strong> Upgrade today to run group-based analytics.
```

**Review by paolodamico — COMMENTED** [review]
Mostly lgtm! Just one comment re the intro tab seems like it's particularly worth fixing

File: frontend/src/scenes/groups/GroupsTabs.tsx (hunk 1 above)
Comment (macobo) [reply frontend/src/scenes/groups/GroupsTabs.tsx:15]: showGroupsOptions is "fixed" in the other PR to account for when we should show upsell as well.

Complicated business logic leads to complicated code. :shrug:

**Comment by macobo** [comment]
Given changing this is now kind of painful and everything but available_features is feature-flagged, merging both of my PRs and making another PR on top of master fixing outlined issues.

File: frontend/src/scenes/groups/GroupsIntroduction.tsx (hunk 3 above)
Comment (macobo) [reply frontend/src/scenes/groups/GroupsIntroduction.tsx:67]: This doesn't work:

image

This is not close to the figma design, so keeping h2

**Pull request merged**